src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -350,7 +351,6 @@ private static object BindToCollection(Type type, IConfiguration config, BinderO
 
                 // Leaf nodes are always reinitialized
                 bindingPoint.TrySetValue(convertedValue);
-                return;
```

Comment:
Was this a mistake in the current code? I don't understand why it can be removed (and if it can, why it was valid before this change).

Reply:
I think I added to verify an assumption I made in the code. All of the tests still passed, which satisfied my assumption. But I'm not 100% sure that there's not an edge case where this would break something. After the recent rebase, this has gone, so I won't reinstate it.

Reply:
I think I would add this line back. The reason being this is handling "leaf nodes". It doesn't make sense to continue processing / checking more things for "leaf nodes".

There doesn't appear to be a functional bug here either way, since the next line does a check for "GetChildren().Any()", which leaf nodes won't have children. So that line gets skipped. However, if more code was added to this method later, it has the potential for a future bug.

Since this change is unrelated to the current PR, I would revert this change.

Reply:
Would you mind reverting for me as I'm away on holiday this week. Many thanks

Reply:
> > After the recent rebase, this has gone, so I won't reinstate it.
> 
> This is still present in the diff, if you didn't intend to remove I think we should move it back.
> 
> > All of the tests still passed, which satisfied my assumption. 
> 
> Do we have a test satisfying this condition? FYI, you could use /p:coverage=true to check for test code coverage in `ConfigurationBinder`

Thanks for the tip and for spotting the issue. It should be reverted, but I'm away this week.

Reply:
Done.